internal/app/machined/pkg/controllers/cri/seccomp_profile_file_test.go

```diff
@@ -0,0 +1,106 @@
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/talos-systems/go-retry/retry"
+
+	criseccompresource "github.com/talos-systems/talos/pkg/machinery/resources/cri"
+	runtimeres "github.com/talos-systems/talos/pkg/machinery/resources/runtime"
+)
+
+func (suite *CRISuite) TestReconcileSeccompProfileFile() {
+	// need to mock mountStatus so that the controller moves ahead with the actual code
+	mountStatus := runtimeres.NewMountStatus(runtimeres.NamespaceName, "EPHEMERAL")
+	suite.Require().NoError(suite.State().Create(suite.Ctx(), mountStatus))
+
+	for _, tt := range []struct {
+		seccompProfileName  string
+		seccompProfileValue map[string]interface{}
+	}{
+		{
+			seccompProfileName: "audit.json",
+			seccompProfileValue: map[string]interface{}{
+				"defaultAction": "SCMP_ACT_LOG",
+			},
+		},
+		{
+			seccompProfileName: "deny.json",
+			seccompProfileValue: map[string]interface{}{
+				"defaultAction": "SCMP_ACT_ERRNO",
+			},
+		},
+	} {
+		seccompProfiles := criseccompresource.NewSeccompProfile(tt.seccompProfileName)
+		seccompProfiles.TypedSpec().Name = tt.seccompProfileName
+		seccompProfiles.TypedSpec().Value = tt.seccompProfileValue
+		suite.Require().NoError(suite.State().Create(suite.Ctx(), seccompProfiles))
+
+		suite.AssertWithin(1*time.Second, 100*time.Millisecond, func() error {
+			if _, err := os.Stat(suite.seccompProfileDirectory + "/" + tt.seccompProfileName); err != nil {
+				if errors.Is(err, os.ErrNotExist) {
+					return retry.ExpectedError(err)
+				}
+
+				return err
+			}
+
+			seccompProfileContent, err := os.ReadFile(suite.seccompProfileDirectory + "/" + tt.seccompProfileName)
+			suite.Assert().NoError(err)
+
+			expectedSeccompProfileContent, err := json.Marshal(tt.seccompProfileValue)
+			suite.Assert().NoError(err)
+
+			suite.Assert().Equal(seccompProfileContent, expectedSeccompProfileContent)
+
+			return nil
+		})
+	}
+
+	// create a directory and file manually in the seccomp profile directory
+	// ensure that the controller deletes the manually created directory/file
+	// also ensure that an update doesn't update existing files timestamp
+	suite.Assert().NoError(os.Mkdir(suite.seccompProfileDirectory+"/test", 0o755))
+	suite.Assert().NoError(os.WriteFile(suite.seccompProfileDirectory+"/test.json", []byte("{}"), 0o644))
+
+	auditJSONSeccompProfile, err := os.Stat(suite.seccompProfileDirectory + "/audit.json")
+	suite.Assert().NoError(err)
+
+	// delete deny.json resource
+	suite.Assert().NoError(suite.State().Destroy(suite.Ctx(), resource.NewMetadata(criseccompresource.NamespaceName, criseccompresource.SeccompProfileType, "deny.json", resource.VersionUndefined)))
+
+	suite.AssertWithin(1*time.Second, 100*time.Millisecond, func() error {
+		auditJSONSeccompProfileAfterUpdate, err := os.Stat(suite.seccompProfileDirectory + "/audit.json")
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				return retry.ExpectedError(err)
+			}
+
+			return err
+		}
+
+		suite.Eventually(func() bool {
```

Comment:
just checking the file seems to fail and had to use `suite.Eventually`. Seems using `filepath.WalkDir` is a little slow